src/shape/polyline.rs

```diff
@@ -143,6 +143,55 @@ impl Polyline {
         }
     }
 
+    /// Extracts the connected components of this polyline, consuming `self`.
+    ///
+    /// Assumes that:
+    /// - This polyline is closed with `self.indices[i][1] == self.indices[(i + 1) % component_length][0]`
+    /// for each component, where `component_length` is the number of vertices in that component.
+    /// - The indices for each component are found contiguously in the index buffer.
```

Comment:
This is technically a corollary of the previous statement, but I think including it improves clarity.